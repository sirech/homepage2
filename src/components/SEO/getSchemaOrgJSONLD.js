export default ({
  isBlogPost,
  url,
  siteUrl,
  title,
  author,
  image,
  siteTitle,
  description,
  datePublished,
}) => {
  const schemaOrgJSONLD = [
    {
      '@context': 'http://schema.org',
      '@type': 'WebSite',
      url,
      name: title,
      alternateName: siteTitle,
    },
  ]

  const post = {
    '@context': 'http://schema.org',
    '@type': 'BlogPosting',
    url,
    name: title,
    alternateName: siteTitle,
    headline: title,
    description,
    author: {
      '@type': 'Person',
      name: author,
    },
    publisher: {
      '@type': 'Organization',
      url: siteUrl,
      name: author,
    },
    mainEntityOfPage: {
      '@type': 'WebSite',
      '@id': siteUrl,
    },
    datePublished,
  }

  if (image) {
    post.image = image
  }

  return isBlogPost
    ? [
        ...schemaOrgJSONLD,
        {
          '@context': 'http://schema.org',
          '@type': 'BreadcrumbList',
          itemListElement: [
            {
              '@type': 'ListItem',
              position: 1,
              item: {
                '@id': url,
                name: title,
              },
            },
          ],
        },
        post,
      ]
    : schemaOrgJSONLD
}
