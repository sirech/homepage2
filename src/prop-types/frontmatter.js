import PropTypes from 'prop-types'

export default PropTypes.shape({
  categories: PropTypes.arrayOf(PropTypes.string).isRequired,
  date: PropTypes.string.isRequired,
  layout: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  draft: PropTypes.bool.isRequired,
  description: PropTypes.string,
}).isRequired
