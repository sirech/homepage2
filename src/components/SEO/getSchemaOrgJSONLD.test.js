import getSchemaOrgJSONLD from './getSchemaOrgJSONLD'

import frontmatter from '../../fixtures/frontmatter'
import siteMetadata from '../../fixtures/siteMetadata'

describe('getSchemaOrgJSONLD', () => {
  let post
  let site

  beforeEach(() => {
    post = {
      html: '<p>this is the content</p>',
      frontmatter: frontmatter(),
    }

    site = {
      siteMetadata: siteMetadata(),
    }
  })

  it('creates the correct structure', () => {
    expect(
      getSchemaOrgJSONLD({
        isBlogPost: true,
        url: post.frontmatter.path,
        title: post.frontmatter.title,
        siteUrl: site.siteMetadata.url,
        author: site.siteMetadata.author,
        image: '/static/logo.png',
        siteTitle: site.siteMetadata.title,
        description: post.frontmatter.description,
        datePublished: post.frontmatter.date,
      })
    ).toMatchSnapshot()
  })
})
