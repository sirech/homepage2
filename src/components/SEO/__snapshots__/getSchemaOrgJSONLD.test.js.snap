// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSchemaOrgJSONLD creates the correct structure 1`] = `
Array [
  Object {
    "@context": "http://schema.org",
    "@type": "WebSite",
    "alternateName": "site title",
    "name": "My Post",
    "url": "/the-post",
  },
  Object {
    "@context": "http://schema.org",
    "@type": "BreadcrumbList",
    "itemListElement": Array [
      Object {
        "@type": "ListItem",
        "item": Object {
          "@id": "/the-post",
          "name": "My Post",
        },
        "position": 1,
      },
    ],
  },
  Object {
    "@context": "http://schema.org",
    "@type": "BlogPosting",
    "alternateName": "site title",
    "author": Object {
      "@type": "Person",
      "name": "me",
    },
    "datePublished": "2018/03/01",
    "description": "this is the description",
    "headline": "My Post",
    "image": "/static/logo.png",
    "mainEntityOfPage": Object {
      "@id": "http://example.com",
      "@type": "WebSite",
    },
    "name": "My Post",
    "publisher": Object {
      "@type": "Organization",
      "name": "me",
      "url": "http://example.com",
    },
    "url": "/the-post",
  },
]
`;
